subprojects/docs/src/docs/userguide/buildEnvironment.adoc

```diff
@@ -13,135 +13,216 @@
+
+.Configuring an HTTP proxy using `gradle.properties`
+====
+[source,properties]
+----
 systemProp.http.proxyHost=www.somehost.org
 systemProp.http.proxyPort=8080
 systemProp.http.proxyUser=userid
 systemProp.http.proxyPassword=password
 systemProp.http.nonProxyHosts=*.nonproxyrepos.com|localhost
-
 ----
-
-
 ====
 
 There are separate settings for HTTPS.
 
-.Configuring an HTTPS proxy
+.Configuring an HTTPS proxy using `gradle.properties`
 ====
-
-`gradle.properties`
-
 [source,properties]
 ----
-
 systemProp.https.proxyHost=www.somehost.org
 systemProp.https.proxyPort=8080
 systemProp.https.proxyUser=userid
 systemProp.https.proxyPassword=password
 systemProp.https.nonProxyHosts=*.nonproxyrepos.com|localhost
-
 ----
-
-
 ====
 
-We could not find a good overview for all possible proxy settings. One place to look are the constants in a file from the Ant project. Here's a https://git-wip-us.apache.org/repos/asf?p=ant.git;a=blob;f=src/main/org/apache/tools/ant/util/ProxySetup.java;hb=HEAD[link] to the repository. The other is a http://download.oracle.com/javase/7/docs/technotes/guides/net/properties.html[Networking Properties page] from the JDK docs. If anyone knows of a better overview, please let us know via the mailing list.
+You may need to set other properties to access other networks. Here are 2 references that may be helpful:
 
+* link:https://git-wip-us.apache.org/repos/asf?p=ant.git;a=blob;f=src/main/org/apache/tools/ant/util/ProxySetup.java;hb=HEAD[ProxySetup.java in the Ant codebase]
+* link:http://download.oracle.com/javase/7/docs/technotes/guides/net/properties.html[JDK 7 Networking Properties]
 
-[[sub:ntlm_authentication]]
 ==== NTLM Authentication
 
 If your proxy requires NTLM authentication, you may need to provide the authentication domain as well as the username and password. There are 2 ways that you can provide the domain for authenticating to a NTLM proxy:
 
 * Set the `http.proxyUser` system property to a value like `__domain__/__username__`.
 * Provide the authentication domain via the `http.auth.ntlm.domain` system property.
 
+[[sec:configuring_task_using_project_properties]]
+=== Configuring a task using project properties
+
+It's possible to change the behavior of a task based on project properties specified at invocation time.
+
+Suppose you'd like to ensure release builds are only triggered by CI. A simple way to handle this is through an `isCI` project property.
+
+++++
+<sample id="configureTaskUsingProjectProperty" dir="userguide/tutorial/configureTaskUsingProjectProperty" title="Prevent releasing outside of CI">
```

Comment:
Users may ask themselves what the difference between extra properties and project properties is in the code example. It is somewhat confusing in terms of terminology.

Reply:
Right, good catch

Reply:
Thanks. Looking at it again I'd do either one of those things:

1. Define the extra property just for the task domain object (assuming that's the only place it is being used).
2. Just say `if (project.hasProperty("isCI"))` then you don't actually need to declare an extra property and explain it in this context.

Reply:
I went for option 2